components/kyma-environment-broker/internal/process/upgrade_kyma/vmuser_upgrade.go

```diff
@@ -0,0 +1,100 @@
+package upgrade_kyma
+
+import (
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker"
+	kebError "github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/error"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/vmauthoverrides"
+	"github.com/sirupsen/logrus"
+
+	"helm.sh/helm/v3/pkg/release"
+)
+
+const (
+	ProfileName = "values"
+)
+
+type HelmClient interface {
+	ListReleases(name string) ([]*release.Release, error)
+	UpgradeRelease(chartDir, releaseName string, namespace string, overrides map[string]interface{}, profile string) (*release.Release, error)
+}
+
+type VMUpgradeProvisionStep struct {
+	operationManager *process.UpgradeKymaOperationManager
+	client           HelmClient
+	vmAuthConfig     vmauthoverrides.Config
+}
+
+func NewVMUpgradeProvisionStep(os storage.Operations, client HelmClient, vmAuthConfig vmauthoverrides.Config) *VMUpgradeProvisionStep {
+	return &VMUpgradeProvisionStep{
+		operationManager: process.NewUpgradeKymaOperationManager(os),
+		client:           client,
+		vmAuthConfig:     vmAuthConfig,
+	}
+}
+
+func (s *VMUpgradeProvisionStep) Name() string {
+	return "Upgrade_VM_User_Overrides"
+}
+
+func (s *VMUpgradeProvisionStep) Run(operation internal.UpgradeKymaOperation, log logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
+	releaseName := operation.InstanceDetails.ShootName
+	isPresent := false
+	releases, err := s.client.ListReleases(s.vmAuthConfig.Namespace)
+	for _, rel := range releases {
+		if rel.Name == releaseName {
+			isPresent = true
+		}
+	}
+	if err != nil {
+		return s.handleError(operation, err, "err while getting release", log)
+	}
+	if !isPresent {
+		log.Info("Release no longer exists. Skip unistallation.")
```

Comment:
If the monitoring doesn't exist yet for the runtime, we should install it. This is how we will roll out for existing runtimes.